ubiquity/dell-bootstrap.py

```diff
@@ -1226,6 +1226,15 @@ def install(self, target, progress, *args, **kwargs):
                 if not found:
                     wfd.write("GRUB_DISABLE_OS_PROBER=true\n")
 
+            #set default recovery_type of 99_dell_recovery grub as 'hdd' for non-Wyse platforms
+            recovery_type = 'hdd'
+            #if wyse mode is on (dell-recovery/mode == 'wyse'), set the recovery_type to be 'factory'
```

Comment:
Where is `dell-recovery/mode` defined?  I don't see it in the templates: https://github.com/dell/dell-recovery/blob/master/debian/dell-recovery.templates

Also where is it set to Wyse?  Exclusively in factory process? Even if it's exclusively in factory process it's probably also useful to declare it and detect Wyse elsewhere (initramfs, earlier in bootstrap etc) so that hand installs from media will DTRT as long as SMBIOS data is accurate.

Reply:
Thanks for the reminder.  I was wondering if any place to list all preseed meaningful to dell-recovery, and now I know it.

Since Wyse platforms will support OEMR feature, we couldn't rely on SMBIOS string to enable Wyse mode.  Currently, for factory process, my plan is to have dell-recovery/mode being set to "wyse" as part of Wyse images.  For a manual install from media, the wyse mode could be activated as long as dell-recovery/mode is set to "wyse" in dell-recovery.seed. (This is the way how I tested my changes.)

In addition to "wyse", could you envision any other scenarios/values we could utilize dell-recovery/mode in the future?  If no, do you think if it makes sense to change dell-recovery/mode string preseed to dell-recovery/wyse_mode boolean preseed?

Reply:
It would be good to document this in the future on wiki or somewhere else more discoverable.
OK then I guess you don't need to automatically set it.

I think boolean is OK for this case.